services/director-v2/src/simcore_service_director_v2/modules/osparc_variables/_api_auth.py

```diff
@@ -40,34 +39,36 @@ async def _get_or_create_for(
     )
 
 
-async def get_or_create_user_api_key(
+async def create_user_api_key(
     app: FastAPI,
     product_name: ProductName,
     user_id: UserID,
 ) -> str:
-    data = await _get_or_create_for(
+    data: ApiKeyGet = await _create_for(
         app,
         product_name=product_name,
         user_id=user_id,
     )
-    return cast(str, data.api_key)
+    assert data.api_key
+    return data.api_key
 
 
-async def get_or_create_user_api_secret(
+async def create_user_api_secret(
     app: FastAPI,
     product_name: ProductName,
     user_id: UserID,
 ) -> str:
-    data = await _get_or_create_for(
+    data: ApiKeyGet = await _create_for(
```

Comment:
why do you need to cast the type here?

Reply:
What @sanderegg means is that ` data: ApiKeyGet = await _create_for(` is equilvanet to  ` data = cast(ApiKeyGet , await _create_for( ...` 
You are forcing mypy to accept this type while it might not be the case
We missed many problems because of this in the past